app/services/force/listing_service.rb

```diff
@@ -122,5 +124,33 @@ def self.array_sort!(listing)
       end
       listings
     end
+
+    private_class_method def self.add_cloudfront_urls_for_listing_images(listings)
+      listing_images = ListingImage.all
+
+      listings.each do |listing|
+        next unless listing['Listing_Images']&.length&.positive?
+
+        listing_images_array = listing_images.select do |li|
+          li.salesforce_listing_id == listing['Id']
+        end
+
+        multiple_listing_images = set_cloudfront_url(listing['Listing_Images'],
+                                                     listing_images_array)
+        listing['Listing_Images'] = multiple_listing_images
+      end
+      listings
+    end
+
+    private_class_method def self.set_cloudfront_url(sf_listing_images, cf_listing_images)
+      sf_listing_images.each do |li|
+        cf_listing_image = cf_listing_images.select do |cf_li|
```

Comment:
If you follow the recommendation I made for `self.add_cloudfront_urls_for_listing_images`, then the `cf_listing_images` argument will be a chain-able `ActiveRecord` relation, which allows us to make this code more compact:
```ruby
cf_listing_image = cf_listing_images.where(raw_image_url: li['Image_URL']).first
```

Reply:
addressed